nolibc/src/lib.rs

```diff
@@ -18,6 +18,42 @@ pub use x86_64::*;
 
 // TODO: These functions are naive implementations that are not performant.
 
+/// Buffer used by readv() and writev() syscalls
+#[repr(C)]
+pub struct Iovec<'a, T: 'a> {
+    /// Buffer start address
+    pub base: *mut T,
+
+    /// Number of bytes to transfer
+    pub size: usize,
+
+    phantom: core::marker::PhantomData<&'a T>,
+}
```

Comment:
I need to learn more about this, obviously. I thought it would tie the lifetime to the `base`. For now, Rust won't let me compile it without the `T` in the struct parameters: `pub struct Iovec<'a, T>` since it's required in `PhantomData`. I changed the `T` in the base to a `u8`.